### ISSUE TITLE
Improve `prefer-set-has`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1338]
Currently, we only report cases that `Array` only called `.includes()`. But in some other cases, it should still be preferred.

```js
const foo = [1, 2, 3];

if (foo.includes(1));

// Still work the same if we use `Set`
for (const element of foo);
const bar = [...foo, somethingElse];
call(...foo);
const length = foo.length; // Use `.size` when switch to Set

// Even this, if the callback don't use the second parameter, they are the same
foo.forEach(element => call(element));
```

We only need make sure the original `Array` have unique values.

**Comment by sindresorhus** [comment]
This is accepted.

**Comment by omril1** [comment]
May I suggest to rename the issue title to something like "reduce false negative"? it's a bit too broad (and overlaps other issues like #2491).
